Speccing fixes

Flix6x opened the pull request.

nhoening commented on timetomodel/utils/debug_utils.py:
```diff
@@ -40,7 +40,10 @@ def render_literal_value(self, value, type_):
-                        [self.render_array_value(x, type_.item_type) for x in value]
+                        [self.render_array_value(
+                            x,
+                            type_.item_type if hasattr(type_, "item_type") else type_
```
Awesome, I did this very same fix locally and was about to make a PR :)

Flix6x commented:
Aside some small fixes, I moved up the trigger to resample, to before data quality is checked.

Why? I ran into a situation where I got an exception on the data time range not reaching all the way to 23:55, but instead stopping at 23:00. I was trying to make 5-minute predictions with an hourly regressor whose last index was at 23:00. I wrote a resampler that  behaves nicely with respect to the right side of the time range upon upsampling (unlike the default pandas resampler used by timetomodel, which strictly interpolates), such that resampling to 5 minutes made the index go all the way to 23:55. However, the exception was still triggered, because the relevant data quality check was triggered before resampling. This is why I switched the order.

Flix6x commented:
Issue #17 is related.
